### PR TITLE
Menu placement improvements

### PR DESCRIPTION
**Pull request by Outcue**
Menus are now placed at point of origin. 
Basic extents checking is done to keep menu from going out of bound to the right.
Menu width is hardcoded. Width can be calculated once I figure out how to get widest measure string.
Menu does not dismiss when click occurs outside of bounds.

I am mainly looking for idiomatic newspeak coding advice.

### CODE REVIEW

File: HopscotchForHTML5.ns [hunk 1]
```diff
@@ -514,17 +514,38 @@ removeContent ^ <Alien[Element]> = (
...
-	| menuContent <Alien[Div]> |
-	menuContent:: computeContentForMenu: menuSupplier.
-	(menuContent at: 'style') 
-	   at: alignment put: 0;
-	   at: 'display' put: 'block'.
-	visual appendChild: menuContent.
-	^menuContent
+    | menu x left maxLeft menuWidth rightInset |
+    
+    (* TODO: We the populated menu dimensions *)
```
Comment (gbracha) [inline HopscotchForHTML5.ns:523]: We? Presumably the comment was "We should use" etc.

File: HopscotchForHTML5.ns [hunk 2]
```diff
@@ -990,6 +1011,8 @@ class HopscotchShell = (
 	currentPresenterX
         toolbarHolder = document createElement: 'div'.
 	contentHolder = document createElement: 'div'.
+    public activeMenu <Alien[Div]> ::= nil.
```
Comment (gbracha) [inline HopscotchForHTML5.ns:1014]: No need to assign nil. That's the default.

File: HopscotchForHTML5.ns [hunk 3]
```diff
@@ -514,17 +514,38 @@ removeContent ^ <Alien[Element]> = (
...
-	   at: 'display' put: 'block'.
-	visual appendChild: menuContent.
-	^menuContent
+    | menu x left maxLeft menuWidth rightInset |
+    
+    (* TODO: We the populated menu dimensions *)
+    menuWidth:: 180.
+    rightInset: 20.
+
+    menu:: computeContentForMenu: menuSupplier.
```
Comment (gbracha) [inline HopscotchForHTML5.ns:527]: So, I'm commenting to be picky for the sake of conveying idiomatic style, so feel free to ignore; 
 menu is only assigned once, and so could be defined as a read-only variable by initializing it at its point of declaration. LIkewise menuWidth and rightInset, though those may change as you compute the real width I suppose. So one might write
| 
menu = computeContentForMenu: menuSupplier. 
menuWidth = 180. 
rightInset = 20. 
x = visual at: 'offsetLeft'.
left 
maxLeft 
|

I also really don't like re-assigning to variables unless we're in a loop, so I'd prefer

maxLeft:: (window at: 'innerWidth') - menuWidth - rightInset.

At that point, one can set maxLeft at the declaration as well.

ifTrue:ifFalse: is just a method call, so one can write

left:: x > maxLeft ifTrue: [maxLeft asString] ifFalse: [ x as String].
or even

left:: (x min: maxLeft) asString.

| 
menu = computeContentForMenu: menuSupplier. 
menuWidth = 180. 
rightInset = 20. 
x = visual at: 'offsetLeft'.
maxLeft  = (window at: 'innerWidth') - menuWidth - rightInset.
left = (x min: maxLeft) asString.
|
    (menu at: 'style') 
        at: alignment put: 0;
        at: 'position' put: 'absolute';
        at: 'left' put: left,'px';
        at: 'width' put: menuWidth asString,'px';
        at: 'display' put: 'block'.
    visual appendChild: menu.
    shell activeMenu: menu.
    shell activeMenuParent: visual.
    ^menu.

I tend to favor annotating all variable declarations with types, but this is no doubt debatable. If the types are obvious, it's less important._

File: HopscotchForHTML5.ns (hunk 2 above)
Comment (gbracha) [reply HopscotchForHTML5.ns:1014]: Likewise below

File: HopscotchForHTML5.ns (hunk 3 above)
Comment (Outcue) [reply HopscotchForHTML5.ns:527]: Yes, please be brutal. It is what I was hoping for!

**Pull request merged**